src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Deprecation.php

```diff
@@ -52,13 +52,19 @@ class Deprecation
      */
     public function __construct($message, array $trace, $file)
     {
-        $this->trace = $trace;
+        $backtraceTriggerDeprecation = $trace[2] ?? null;
+        if ($backtraceTriggerDeprecation && 'trigger_deprecation' === $backtraceTriggerDeprecation['function']) {
+            $file = $backtraceTriggerDeprecation['file'];
+            unset($trace[1]);
```

Comment:
can be removed if `array_values()` is removed below, isn't it? that would save some memcopies.

Reply:
I'm not sure to understand properly. `unset` is not here because of `array_values`, it's `array_values` that is needed because of `unset`. And `unset` is needed since if I keep the trace of `deprecations-contracts/functions`, `getType` and `getPathType` would mark `indirect` deprecations as `direct` (because `deprecations-contracts` is a dependency of the project).